RFCs/0000-rfc-add-pdk-dubugger.md

```diff
@@ -0,0 +1,167 @@
+
+```
+$ pdk console -e "'true' == true" --run-once
+Ruby Version: 2.5.1
+Puppet Version: 6.7.2
+Puppet Debugger Version: 0.12.3
+Created by: NWOps <corey@nwops.io>
+Type "commands" for a list of debugger commands
+or "help" to show the help screen.
+
+
+1:>> 'true' == true
+ => false
+```
+
+### Within a module
+The other way of invoking the console will be in the context of a module.  However, because the user is inside a module, it will be scoped down to only the module's dependencies inside that module.  This is important because it gives the user a clean sandbox respective of their current dependencies and hiera data. 
+
+Example workflow
+
+1. `cd apache_module`
+2. `pdk cnnsole`
+
+Because we can check if we are in a module via `PDK::Util.in_module_root?` we can make some assumptions and automate a few settings for the user. The settings we can assume are:
+
+1. basemodulepath=./spec/fixtures/modules
+2. hiera_config=./hiera.yaml (at least if inside a control_repo)
+
+*NOTE* This would mean that a `rake spec_prep` would have to be run to download any modules.  
+
+*NOTE* For multi-tier lookups the user would need to override the hiera_config file location. 
+
+The command invoke within a module is still the same.
+
+```
+$ pdk console -e "'true' == true" --run-once
+Ruby Version: 2.5.1
+Puppet Version: 6.7.2
+Puppet Debugger Version: 0.12.3
+Created by: NWOps <corey@nwops.io>
+Type "commands" for a list of debugger commands
+or "help" to show the help screen.
+
+
+1:>> 'true' == true
+ => false
+```
+
+### Additional Config Options
+Many users have used the debugger without understanding what options can be passed in. This is because the debugger is a puppet application.  It inherits all the config options that puppet accepts. So while the puppet debugger does have some options itself most are inherited and passed through to puppet (Same as puppet apply).  See the following options [listed here](https://puppet.com/docs/puppet/5.5/configuration.html). 
+
+With this is mind we will want to respect these options when using the `pdk console` command.  We will need to forward these options to the command so the user can override anything they desire.  By default, the debugger will just read the settings via the config file using `Puppet.settings`. 
+
+### Special Arguments
+There are two config options that are specific to PDK that make the debugger's involvement special.  Without the PDK the debugger cannot switch ruby or puppet versions easily.  The PDK makes this simple using the 
+`--puppet-version=5.5.12` or `--pe-version=2018.1` arguments.  These arguments will allow the user to use the debugger to see how the language changed between versions. 
+
+`pdk console --puppet-version=5.5.12 -e 'true' =~ Boolean'`
+
+`pdk console --puppet-version=6.7.0 -e "'true' =~ Boolean"`
+
+This would tell the user if the code has changed behavior in the newer version.  
+
+### Add to puppet-module-posix- gems
```

Comment:
Hrmm, this would stop merging... what about Windows?

Reply:
Is there a correct name for these gems that addresses both platforms?

Reply:
`Puppet Module Meta-gems`

`puppet-module meta gems`  <--- My pick

`puppet-module-* gems`

Reply:
ping @rodjek  @bmjen  👆👆

Reply:
Fixed with the updated gem name.